src/Raven.Server/Documents/Handlers/DocumentReplicationHandler.cs

```diff
@@ -96,20 +120,44 @@ public class DocumentReplicationRequestHandler : DatabaseRequestHandler
         {
             switch (messageTypeAsString)
             {
+				case Constants.Replication.MessageTypes.Heartbeat:
+					_lastHeartbeatReceive = DateTime.UtcNow; //not sure that this variable is actually needed
+		            _heartbeatTimer.Change(TimeSpan.Zero, HeartbeatTimeout);
+					break;
                 case Constants.Replication.MessageTypes.GetLastEtag:
-                    using (context.OpenReadTransaction())
-                        WriteLastEtagResponse(webSocketStream, context, srcDbId);
-                    break;
+		            try
+		            {
+			            using (context.OpenReadTransaction())
+				            WriteLastEtagResponse(webSocketStream, context, srcDbId);
+		            }
+		            catch (Exception e)
+		            {
+			            Log.Error("Failed to write last etag response; Reason: " + e);
+		            }
```

Comment:
And close connection with the error. We can't recover from something like this, other side had to reconnect